src/components/AllComponents.js

```diff
@@ -201,4 +147,15 @@ function AllComponents() {
   );
 }
 
+const getFilteredCharacters = (allCharacters, userCharacters, filterData) => {
+  const baseChars = filterData.isUserDeck ? userCharacters : allCharacters;
+  return baseChars.filter(character => {
+    return (!filterData.searchTerm || character.name.toLowerCase().includes(filterData.searchTerm.toLowerCase())) &&
```

Comment:
"not filter or check filter" means:
- if filter item not present, ignore it (i.e. get ALL)
- if present, check the filter against the current character

This works because formData returns falsy values if that filter item is not selected